great_expectations/execution_engine/execution_engine.py

```diff
@@ -196,7 +212,7 @@ def __init__(
         self._caching = caching
         # NOTE: 20200918 - this is a naive cache; update.
         if self._caching:
-            self._metric_cache = {}
+            self._metric_cache: Union[Dict, NoOpDict] = {}
```

Comment:
Do you know why we defined `NoOpDict`? Seems like an odd object

Reply:
No, but from context I would guess for cases where we don't want to actually cache/retrieve metrics.